Sources/Confidence/Confidence.swift

```diff
@@ -169,25 +149,155 @@ public class Confidence: ConfidenceEventSender {
+            do {
+                try await self.fetchAndActivate()
+                debugLogger?.logContext(
+                    action: "RemoveContext",
+                    context: newContext)
+            } catch {
+                debugLogger?.logMessage(
+                    message: "Error when removing context key: \(error)",
+                    isWarning: true)
+            }
+        }
+        await awaitReconciliation()
     }
 
     /**
-    Listen to changes in the context that is local to this Confidence instance.
+    Adds/override entry to local context data. Does not trigger fetchAndActivate after the context change.
     */
-    public func contextChanges() -> AnyPublisher<ConfidenceStruct, Never> {
-        return contextSubject
-            .dropFirst()
-            .removeDuplicates()
-            .eraseToAnyPublisher()
+    public func putContextLocal(context: ConfidenceStruct, removeKeys removedKeys: [String] = []) {
+        let newContext = contextManager.updateContext(withValues: context, removedKeys: removedKeys)
+        debugLogger?.logContext(
+            action: "PutContextLocal",
+            context: newContext)
     }
 
-    public func track(eventName: String, data: ConfidenceStruct) throws {
-        try eventSenderEngine.emit(
-            eventName: eventName,
-            data: data,
-            context: getContext()
+    public func putContext(key: String, value: ConfidenceValue) {
+        taskManager.currentTask = Task {
+            await putContextAndWait(key: key, value: value)
+        }
+    }
+
+    public func putContext(context: ConfidenceStruct) {
+        taskManager.currentTask = Task {
+            await putContextAndWait(context: context)
+        }
+    }
+
+    public func putContext(context: ConfidenceStruct, removeKeys removedKeys: [String] = []) {
+        taskManager.currentTask = Task {
+            await putContextAndWait(context: context, removedKeys: removedKeys)
+        }
+    }
+
+    public func removeContext(key: String) {
+        taskManager.currentTask = Task {
+            await removeContextAndWait(key: key)
+        }
+    }
+
+    public func putContext(context: ConfidenceStruct, removedKeys: [String]) {
+        taskManager.currentTask = Task {
+            let newContext = contextManager.updateContext(withValues: context, removedKeys: removedKeys)
+            do {
+                try await self.fetchAndActivate()
+                debugLogger?.logContext(
+                    action: "RemoveContext",
+                    context: newContext)
+            } catch {
+                debugLogger?.logMessage(
+                    message: "Error when putting context: \(error)",
+                    isWarning: true)
+            }
+        }
+    }
+
+    /**
+    Ensures all the already-started context changes prior to this function have been reconciliated
+    */
+    public func awaitReconciliation() async {
```

Comment:
I would suggest `awaitReady()` instead...

Reply:
I think we should keep the same [name used in Kotlin](https://github.com/spotify/confidence-sdk-android/blob/a2b10f1bfb948b37425f8c30c5e9445c0671d83d/Confidence/src/main/java/com/spotify/confidence/Confidence.kt#L66C17-L66C36), i.e. `awaitReconciliation`

I'd be happy to create a pair of "renaming PRs" for both repos after this though, if we want to reach proper alignment (this would make sense especially now that the APIs between platforms are much more similar!)